### PR TITLE
Add preStop hook for sending SIGRTMIN+3 to systemd

### PR DESCRIPTION
**Pull request by avisiedo**
Add lifecycle handler for sending the proper signal to systemd before stopping the container.

### CODE REVIEW

File: config/static/single-pod/pod-main.yaml [hunk 1]
```diff
@@ -144,6 +144,12 @@ spec:
       privileged: false
     terminationMessagePath: /dev/termination-log
     terminationMessagePolicy: File
+    # https://kubernetes.io/docs/tasks/configure-pod-container/attach-handler-lifecycle-event/
+    # https://www.freedesktop.org/software/systemd/man/systemd.html#SIGRTMIN+3
+    lifecycle:
+      preStop:
+        exec:
+          command: ["kill", "-s", "SIGRTMIN+3", "1"]
```
Comment (frasertweedale) [inline config/static/single-pod/pod-main.yaml]: This is super nice and works like a charm.  A "must have" for a systemd-based container.

One nit (extremely minor): Perhaps `["kill", "-RTMIN+3", "1"]`, to save one line in the YAML output from `oc get -o yaml pod/...`.

**Review by frasertweedale — DISMISSED** [review]
One nit.  Feel free to merge with or without my very minor suggestion.

File: config/static/single-pod/pod-main.yaml [hunk 2]
```diff
@@ -144,6 +144,12 @@ spec:
       privileged: false
     terminationMessagePath: /dev/termination-log
     terminationMessagePolicy: File
+    # https://kubernetes.io/docs/tasks/configure-pod-container/attach-handler-lifecycle-event/
+    # https://www.freedesktop.org/software/systemd/man/systemd.html#SIGRTMIN+3
```
Comment (frasertweedale) [inline config/static/single-pod/pod-main.yaml:148]: Thank you for putting references in the comments :ok_hand:

File: config/static/single-pod/pod-main.yaml (hunk 1 above)
Comment (avisiedo) [reply config/static/single-pod/pod-main.yaml]: Updated. Thanks @frasertweedale

File: config/static/single-pod/pod-main.yaml (hunk 2 above)
Comment (avisiedo) [reply config/static/single-pod/pod-main.yaml:148]: you are welcome! :)

**Comment by avisiedo** [comment]
Change updated with the review observations; Merging and closing ticket

**Pull request merged**